bids/layout/layout.py

```diff
@@ -32,6 +32,14 @@ def __init__(self, filename, layout):
         super(BIDSFile, self).__init__(filename)
         self.layout = layout
 
+    def __getattr__(self, attr):
+        # Ensures backwards compatibility with old File_ namedtuple, which is
+        # deprecated as of 0.7.
+        if attr in self.entities:
+            return self.entities[attr]
+        raise AttributeError("%s object has no attribute named %r" %
+                             (self.__class__.__name__, attr))
```

Comment:
This never calls `super()`, which makes me suspect that a lot of its normal entries are going to end up hidden, unless there's some other Python magic I don't know about. What about:

```Python
def __getattr__(self, attr):
    try:
        return super(BIDSFile, self).__getattr__(attr)
    except AttributeError:
        if attr == 'filename':
            warnings.warn("The `file.filename` attribute is deprecated; use `file['path']`.", DeprecationWarning)
            attr = 'path'
        if attr in self.entities:
            warnings.warn("Attribute access is deprecated as of 0.7. Use dictionary access.", DeprecationWarning)
            return self.entities[attr]
        raise
```

Reply:
+1 on super call, but why not to do it after local handling, so to be able to overload what super does?

your patch adds deprecation - is that what is planed/desired ( @tyarkoni  ?)

Reply:
about super... ah -- it was calling/catch super first since grabbit's `File` doesn't even implement `__getattr__` ATM.  So it was for some future proofing @effigies ?

Reply:
Ah, my thought was that we would actually be hiding real attributes, e.g. I suspected the following would prevent `B().d` from being accessed:

```Python
class A:
    d = 1

class B(A):
    def __getattr__(self, attr):
        if attr == 'e':
            return 2
        raise AttributeError(attr)
```

Turns out that proper attributes take precedence. I guess because otherwise you couldn't call `obj.__getattr__` at all.

Reply:
yes, it is a pythonic magic of `__getattr__` vs `__getattribute__`